src/arch/main.cpp

```diff
@@ -52,27 +38,35 @@ int main(int argc, char *arg[]){
 
   checkEndianess();
 
-  startupHook(argc, arg);
+  if(auto result = CForteArchitecture::initialize(argc, arg); result != 0){
+     return result;
+  }
 
-#ifdef FORTE_IO_PLCNEXT
-  sleep(3);
-  DeviceStatus::startup();
-#endif
+  struct callOnExit {
+    int (*onExit)();
+    ~callOnExit() { onExit(); }
+  };
+
+  callOnExit{CForteArchitecture::deinitialize};
 
   hookSignals();
 
   const char *pIpPort = parseCommandLineArguments(argc, arg);
```

Comment:
could you please remove the prefixed `p` here, as the coding guidelines abolished the use of Hungarian notation for most parts

Reply:
done